lighthouse-core/test/audits/splash-screen-test.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+'use strict';
+
+const SplashScreenAudit = require('../../audits/splash-screen');
+const assert = require('assert');
+const manifestParser = require('../../lib/manifest-parser');
+
+const manifestSrc = JSON.stringify(require('../fixtures/manifest.json'));
+const EXAMPLE_MANIFEST_URL = 'https://example.com/manifest.json';
+const EXAMPLE_DOC_URL = 'https://example.com/index.html';
+const exampleManifest = noUrlManifestParser(manifestSrc);
+
+const GatherRunner = require('../../gather/gather-runner.js');
+
+function generateMockArtifacts() {
+  const computedArtifacts = GatherRunner.instantiateComputedArtifacts();
+  const mockArtifacts = Object.assign({}, computedArtifacts, {
+    Manifest: exampleManifest
+  });
+  return mockArtifacts;
+}
+
+/**
+ * Simple manifest parsing helper when the manifest URLs aren't material to the
+ * test. Uses example.com URLs for testing.
+ * @param {string} manifestSrc
+ * @return {!ManifestNode<(!Manifest|undefined)>}
+ */
+function noUrlManifestParser(manifestSrc) {
+  return manifestParser(manifestSrc, EXAMPLE_MANIFEST_URL, EXAMPLE_DOC_URL);
+}
+
+/* eslint-env mocha */
+describe('PWA: splash screen audit', () => {
+  describe('basics', () => {
+    it('fails if page had no manifest', () => {
```

Comment:
it would be nice to move these to be true unit tests, so that the `MultiCheckAudit` subclasses were just getting back a set of checks with `.passing` set to true or false, and the unit test could just check if they respond to the correct passing values. Then the actual manifest verification stuff would live in the test for the manifest computed artifact, where the code for that actually lives.

Would also cut down on some of these duplicated tests (e.g. the one off cases) and minimize things like the manifest parser setup even more